vm/src/object/ext.rs

```diff
@@ -107,6 +107,7 @@ impl<T: PyPayload> std::borrow::ToOwned for PyExact<T> {
     }
 }
 
+/// PyRef but guaranteed not to be a subtype instance
```

Comment:
@DimitrisJim I don't think this single line fulfill your concern, but it is mostly it is.

Other types are roughly tabled in https://github.com/RustPython/RustPython/blob/main/vm/src/object/core.rs#L1-L12

Other types in core and ext are mostly hidden - hopefully

Reply:
Thanks! At some point in the future (after I'm hopefully done documenting the compiler), I'll start documenting as much of the vm as possible. We should set up a call then to discuss all these so I can get the docs right.